### PR TITLE
deprecate /eth2/beacon_chain/req/beacon_blocks_by_{range,root}/1/

### PR DESCRIPTION
**Pull request by tersec**
For when https://github.com/ethereum/consensus-specs/pull/3014 is merged

This could wait until an actual consensus spec release, but doesn't have to, because as noted:
> There haven't been any phase0 blocks in mainnet or any of the still-supported testnets in [`MIN_EPOCHS_FOR_BLOCK_REQUESTS`](https://github.com/ethereum/consensus-specs/blob/dev/specs/phase0/p2p-interface.md#configuration) regardless, so the spec already technically supports not serving said blocks.

So it can be merged as soon as that spec PR is merged.

### CODE REVIEW

**Review by etan-status — APPROVED** [review]
Should we still do the rate limiting?

```nim
     peer.updateRequestQuota(blockByRangeLookupCost)
     peer.awaitNonNegativeRequestQuota()
```

**Comment by tersec** [comment]
> Should we still do the rate limiting?
> 
> ```nim
>      peer.updateRequestQuota(blockByRangeLookupCost)
>      peer.awaitNonNegativeRequestQuota()
> ```

Sure, why not: https://github.com/status-im/nimbus-eth2/pull/4207/commits/f1960c8237690da9f45e0cb649e8d01fa0994293

Discourages use of it, since it's not entirely free.

File: beacon_chain/sync/sync_protocol.nim [hunk 1]
```diff
@@ -274,142 +274,23 @@ p2pProtocol BeaconSync(version = 1,
...
-
-        await response.writeBytesSZ(uncompressedLen, bytes, []) # phase0
-        inc found
-
-    debug "Block root request done",
-      peer, roots = blockRoots.len, count, found
+    # https://github.com/tersec/consensus-specs/blob/deprecate-phase0-beacon-blocks-req-resp/specs/phase0/p2p-interface.md#beaconblocksbyroot
+    # `/eth2/beacon_chain/req/beacon_blocks_by_root/1/` is deprecated. Clients
+    # MAY respond with an empty list during the deprecation transition period.
+    discard
```
Comment (zah) [inline beacon_chain/sync/sync_protocol.nim]:
```suggestion
    peer.updateRequestQuota(blockByRootLookupCost)
    peer.awaitNonNegativeRequestQuota()
```

**Comment by tersec** [comment]
https://github.com/ethereum/consensus-specs/pull/3014 was merged

**Comment by tersec** [comment]
https://github.com/status-im/nimbus-eth2/pull/4378

**Pull request closed without merging**